app/controllers/reports/patient_lists_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Reports::PatientListsController < AdminController
+  skip_after_action :verify_authorized, if: -> { current_admin.permissions_v2_enabled? }
+  after_action :verify_authorization_attempted, if: -> { current_admin.permissions_v2_enabled? }
+
+  def show
+    scope = if current_admin.permissions_v2_enabled?
+      if region_class == "facility_group"
+        authorize_v2 { current_admin.accessible_facility_groups(:view_pii) }
+      else
+        authorize_v2 { current_admin.accessible_facilities(:view_pii) }
+      end
+    else
+      authorize([:cohort_report, region_class.classify.constantize], :patient_list?)
```

Comment:
I can't figure out how to use the pundit system to return the scope of records that the admin has patient_list access to...which is what I want so that we can be consistent w/ the v2 way of doing things here.

Reply:
Since pundit is based off of PERM, it identifies policy checking and scoping as two distinct operations on a resource.

What you are doing here is a policy check / authorization in pundit-parlance, I believe what you need is a scope:

```
policy_scope([:cohort_report, region_class.classify.constantize])
```

And potentially call,

```
policy([:manage, region_class.classify.constantize]).patient_list?
```

...in the views where you want to hide/show the buttons, like: [this](https://github.com/simpledotorg/simple-server/blob/master/app/views/analytics/facilities/show.html.erb#L82) and [this](https://github.com/simpledotorg/simple-server/blob/master/app/views/analytics/districts/show.html.erb#L73).

That should be sufficient.